README.md

```diff
@@ -1,80 +1,98 @@
+```
 
-Rather than having a separate plugin for each kind of content transform you want to be able to do, why not just have one? Parse natural language, markdown, or whatever else you want with a minimalistic and simple interface 🍻
+```js
+const content = require('posthtml-content')
 
-## Install
+const options = { txt: (str) => str.replace(/Windows/g, 'winDOGE') }
 
-```bash
-npm i posthtml-content --save
+posthtml([ content(options) ])
+  .process(html)
+  .then((result) => result.html)
 ```
 
-> **Note:** This project is compatible with node v6+ only
+```html
+<p>Please use WinDOGE 98</p>
+```
+
+Now pass in an object to `posthtml-content`. Each object property represents an attribute that will be searched for in the HTML. The value is a function that will get that element's contents as a string, and replace the contents with whatever string is returned from the function.
 
-## Usage
+The custom attribute will be removed from the element and its contents are replaced with your transformed version.
 
-Start with some html you want to transform in some way. Add an attribute of your choosing to an element that has contents you want to transform.
+You can use any external library (e.g PostCSS, Babel, Marked) within the transform function. Just make sure you are passing in a function that takes a string and returns a string. You might have to wrap the library function if it doesn't behave like this, but it will work with anything that transforms content.
+
+It also possible to load contents from an external file, which gets transformed and applied as the elements content.
 
 ```html
-<p windoge>Please use windows 98</p>
+<article txt="lorem.txt"></article>
 ```
 
-Now pass in an object to `posthtml-content`. Each key in the object represents an attribute that will be searched for in the html. The value is a function that will get that element's contents as a string, and replace the contents with whatever string is returned from the function.
-
 ```js
-const content = require('posthtml-content')({
-  windoge: (str) => str.replace(/windows/g, 'winDOGE')
-})
+const content = require('posthtml-content')
 
-posthtml([plugin]).process(html)
-```
+const options = { txt: (file) => file.toUpperCase() }
 
-The plugin will remove the custom attribute from the element and replace its contents with your transformed version. Wow!
+posthtml([ content(options) ])
+  .process(html)
+  .then((result) => result.html)
+```
 
 ```html
-<p>Please use winDOGE 98</p>
+<artcile>LOREM IPSUM DOLOR SIT AMET,...</article>
 ```
 
-If you return an [A+ compliant promise](https://promisesaplus.com/) from your content function, it will resolve and work in your templates as well.
-
-You can use external libraries for this as well, no problem. Just make sure you are passing in a function that takes a string and returns a string. You might have to wrap the library function if it doesn't behave like this, but it will work with anything that transforms content.
```

Comment:
Promises are supported, just removed the link to PromiseA+ spec site, but yes maybe improve docs here, that returning a promise is supported.